src/main/java/edu/ie3/datamodel/io/processor/timeseries/TimeSeriesProcessor.java

```diff
@@ -0,0 +1,313 @@
+    Map<String, FieldSourceToMethod> entryMapping =
+        mapFieldNameToGetter(entryClass, Collections.singletonList("value")).entrySet().stream()
+            .collect(
+                Collectors.toMap(
+                    Map.Entry::getKey, entry -> new FieldSourceToMethod(ENTRY, entry.getValue())));
+    Map<String, FieldSourceToMethod> valueMapping;
+    if (!valueClass.equals(WeatherValue.class)) {
+      valueMapping =
+          mapFieldNameToGetter(valueClass).entrySet().stream()
+              .collect(
+                  Collectors.toMap(
+                      Map.Entry::getKey,
+                      entry -> new FieldSourceToMethod(VALUE, entry.getValue())));
+    } else {
+      /* Treat the nested weather values specially. */
+      /* Register all getters of the weather value itself */
+      valueMapping =
+          new HashMap<>(
+              mapFieldNameToGetter(valueClass, Arrays.asList("irradiation", "temperature", "wind"))
+                  .entrySet().stream()
+                  .collect(
+                      Collectors.toMap(
+                          Map.Entry::getKey,
+                          entry -> new FieldSourceToMethod(VALUE, entry.getValue()))));
+      /* Register all getters of the nested irradiation value */
+      valueMapping.putAll(
+          (mapFieldNameToGetter(IrradiationValue.class).entrySet().stream()
+              .collect(
+                  Collectors.toMap(
+                      Map.Entry::getKey,
+                      entry -> new FieldSourceToMethod(WEATHER_IRRADIATION, entry.getValue())))));
+      /* Register all getters of the nested irradiation value */
+      valueMapping.putAll(
+          (mapFieldNameToGetter(TemperatureValue.class).entrySet().stream()
+              .collect(
+                  Collectors.toMap(
+                      Map.Entry::getKey,
+                      entry -> new FieldSourceToMethod(WEATHER_TEMPERATURE, entry.getValue())))));
+      /* Register all getters of the nested irradiation value */
+      valueMapping.putAll(
+          (mapFieldNameToGetter(WindValue.class).entrySet().stream()
+              .collect(
+                  Collectors.toMap(
+                      Map.Entry::getKey,
+                      entry -> new FieldSourceToMethod(WEATHER_WIND, entry.getValue())))));
+    }
+
+    /* Put everything together */
+    HashMap<String, FieldSourceToMethod> jointMapping = new HashMap<>();
+    jointMapping.putAll(timeSeriesMapping);
+    jointMapping.putAll(entryMapping);
+    jointMapping.putAll(valueMapping);
+
+    /* Let uuid be the first entry */
+    return Collections.unmodifiableSortedMap(putUuidFirst(jointMapping));
+  }
+
+  @Override
+  public Optional<LinkedHashMap<String, String>> handleEntity(TimeSeries entity) {
+    throw new UnsupportedOperationException(
+        "Don't invoke this simple method, but TimeSeriesProcessor#handleTimeSeries(TimeSeries).");
```

Comment:
Okay this should definitely not be implemented that way!  When implemented this way, this makes the whole interface incompatible with all other implementations. 

Example: Someone writes a class that can handle EntityProcessors and now, by default calls `handleEntity` this will cause this person a lot of headaches ...   😨

Reply:
just dived a little bit more into this and I somehow doubt that the `Processor` in it's current implementation is capable of doing what we want to achieve here due to the differences in output (`Set<LinkedHashMap<String,String>>` vs. `LinkedHashMap<String,String>`). Im wondering if this is somehow solvable in the current concept or if there is no other way to deal with this ....🤔